internal/jptrace/translator.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jptrace
+
+import (
+	otlp2jaeger "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+func ProtoFromTraces(traces ptrace.Traces) []*model.Batch {
+	batches := otlp2jaeger.ProtoFromTraces(traces)
+	for i, batch := range batches {
```

Comment:
Seems unsafe, is it guaranteed that batches and resources have identical cardinality?

Reply:
@yurishkuro I was following the logic in the upstream translator: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/jaeger/traces_to_jaegerproto.go#L25

Reply:
but how do we know upstream logic won't change? A robust way would be to build a map of spanId->*model.Span first and then loop through ptrace spans, find corresponding model.Span and transfer attributes into warnings.

btw, we should probably delete those attributes from model.Span, since ptrace would just copy them.